src/actions/WalletListMenuActions.js

```diff
@@ -89,10 +91,36 @@ export function walletListMenuAction(walletId: string, option: WalletListMenuKey
       return (dispatch: Dispatch, getState: GetState) => {
         const state = getState()
         const { currencyWallets } = state.core.account
-        const wallet = currencyWallets[walletId]
-        const xPub = wallet.displayPublicSeed
-        const xPubExplorer = wallet.currencyInfo.xpubExplorer && xPub ? sprintf(wallet.currencyInfo.xpubExplorer, xPub) : ''
-        dispatch({ type: 'OPEN_VIEWXPUB_WALLET_MODAL', data: { xPub, walletId, xPubExplorer } })
+        const { displayPublicSeed, currencyInfo } = currencyWallets[walletId]
+        const { xpubExplorer } = currencyInfo
+
+        Airship.show(bridge => (
+          <ButtonsModal
+            bridge={bridge}
+            buttons={
+              xpubExplorer != null
+                ? {
+                    copy: { label: s.strings.fragment_request_copy_title, type: 'secondary' },
+                    link: { label: s.strings.transaction_details_show_advanced_block_explorer, type: 'secondary' }
+                  }
+                : {
+                    copy: { label: s.strings.fragment_request_copy_title, type: 'secondary' }
+                  }
```

Comment:
Create consts and merge the right ones together.

Reply:
Done, much nicer.